File: app-core/src/main/java/kr/flab/wiki/core/login/repository/UserLoginRepositoryImpl.java
package kr.flab.wiki.core.login.repository;

import kr.flab.wiki.core.login.persistence.User;

import java.util.HashMap;

public class UserLoginRepositoryImpl implements UserLoginRepository{
    HashMap<String, String> database = new HashMap<>();
    @Override
    public boolean findByIdWithPassword(User user) {
        String password = database.get(user.getEmail());
        if(password != null){
            return isPasswordEquals(password, user.getPassword());
        }
        return false;
    }

    private boolean isPasswordEquals(String origin, String attempt) {
        return origin.equals(attempt);
    }
}

File: app-core/src/main/kotlin/kr/flab/wiki/core/login/repository/UserLoginRepositoryImpl.kt
package kr.flab.wiki.core.login.repository

import kr.flab.wiki.core.login.persistence.User
import kr.flab.wiki.core.login.persistence.UserEntity

class UserLoginRepositoryImpl : UserLoginRepository {
    private val database : HashMap<String, String> = HashMap()

    override fun isUserExists(email: String, password: String): Boolean {
        val originPassword : String = database[email] ?: return false
        return originPassword == password
    }

    override fun findUserWithIdAndPassword(email: String, password: String): User? {
        if(isUserExists(email, password)){
            return UserEntity().apply {
                this.email = email
                this.password = password
            }
        }
        return null
    }
}
